docs/user/troubleshooting.md

```diff
@@ -167,51 +167,41 @@ You can either increase the capacity of your backend or reduce the volume of log
 - Reduce emitted data by re-configuring the pipeline (for example, by disabling certain inputs or applying namespace filters).
 - Reduce the amount of log data generated by your applications.
 
-## EOF Error When Exporting OTLP Data
+## Transform or Filter Rule Has No Effect
 
 ### Symptom
 
-The Telemetry gateway logs show errors like: `OTLP export failed: ... rpc error: code = Unavailable desc = error reading from server: EOF`.
+You have configured a `transform` or `filter` section in your pipeline, but the data arriving at your backend is not modified, or data you expect to be dropped is still present.
 
 ### Cause
 
-An "End of File" (EOF) error indicates that the connection to your OTLP backend was closed unexpectedly by the server. This can be caused by several issues:
-* **Incorrect Endpoint Address**: The gateway is connecting to the wrong host or port.
-* **TLS Configuration Mismatch**: The gateway's TLS settings (`insecure`, `ca`, `cert`, `key`) do not match what the backend server expects. For example, you are connecting without TLS (`insecure: true`) to a server that requires it.
-* **Network Policy**: A Kubernetes `NetworkPolicy` is blocking egress traffic from the `telemetry-operator-gateway` pods to the OTLP endpoint.
-* **Backend Unavailability**: The backend service is crashing, restarting, or is otherwise unhealthy, causing it to drop active connections.
+This usually happens for one of the following reasons:
+* **Incorrect Execution Order**: You're filtering data based on a field's original value, but a transformation rule has already changed it. Transformation rules always run before filter rules.
+* **Condition Never Met**: The condition in your rule is valid (otherwise, the pipeline condition `ConfigurationGenerated` would have a `False` status with the reason `OTTLSpecInvalid`), but never finds a match in the data. This is often due to a case-sensitive value mismatch or a flawed regular expression.
```

Comment:
```suggestion
* **Condition Never Met**: The condition in your rule is valid (otherwise, you'd see pipeline condition `ConfigurationGenerated`:`False` with the reason `OTTLSpecInvalid`), but never finds a match in the data. This is often due to a case-sensitive value mismatch or a flawed regular expression.
```